### DISCUSSION TITLE
There was an error in your GraphQL query: Cannot return null for non-nullable field WordpressWpProduct.databaseId

### DISCUSSION THREAD

**Discussion by maghaffar** [gatsbyjs/gatsby#38903] [Help]
In my Wordpress I am attaching a Product to my Post and I am building my site using Gatsby
When that product is deleted and Netlify builds again it gives the following error:
**_There was an error in your GraphQL query:
Cannot return null for non-nullable field WordpressWpProduct.databaseId.
The field "WordpressWpProduct.databaseId." was explicitly defined as non-nullable via the schema customization API (by yourself or a plugin/theme). This means that this field is not optional and you have to define a value. If this is not your desired behavior and you defined the schema yourself, go to "createTypes" in gatsby-node._**

In order to resolve this I have to Clear Cache on Netlify everytime. Is there any way to solve this problem?

**Comment by violy** [comment]
If clearing your cache solve the problem, it's probably due to an access to the cache himself … 
- Did you have this error only in develop mode or also at build time ?
- Did you have this error also in GraphiQL ?

Can you share your query ?

**Reply by maghaffar** [reply]
Once a build is completed and site is live and then we delete the product from wordpress
Then the site is built again using the cache and it gives the error
This also occurs during build time on netlify
This is the query
fragment Product on WordpressWpProduct {
    id
    title
    slug
    date
    images
    strainType
    brand_name
    databaseId
    }
